scans/worker.py

```diff
@@ -0,0 +1,80 @@
+import os
+import json
+
+from scans import decimalencoder
+import boto3
+dynamodb = boto3.resource('dynamodb')
+from scans import Response
+
+
+def worker(event, context):
+    table = dynamodb.Table(os.environ['DYNAMODB_TABLE'])
+
+    # Example event to play with
+    # {
+    #     "Records": [
+    #         {
+    #             "eventID": "1",
+    #             "eventVersion": "1.0",
+    #             "dynamodb": {
+    #                 "Keys": {
+    #                     "id": {
+    #                         "S": "a88cb45c-0ac1-47b3-865b-a92f05d7612b"
+    #                     }
+    #                 },
+    #                 "NewImage": {
+    #                     "target": {
+    #                         "S": "ssh.mozilla.com"
+    #                     },
+    #                     "port": {
+    #                         "I": 22
+    #                     },
+    #                     "id": {
+    #                         "S": "a88cb45c-0ac1-47b3-865b-a92f05d7612b"
+    #                     }
+    #                 },
+    #                 "StreamViewType": "NEW_IMAGE",
+    #                 "SequenceNumber": "111",
+    #                 "SizeBytes": 128
+    #             },
+    #             "awsRegion": "us-west-2",
+    #             "eventName": "INSERT",
+    #             "eventSourceARN": "event:source:ARN",
+    #             "eventSource": "aws:dynamodb"
+    #         }
+    #     ]
+    # }
+
+    for record in event["Records"]:
+        if not record["eventName"] == "INSERT":
+            continue
+
+        item = record["dynamodb"]["NewImage"]
+        scan_id = item["id"]["S"]
+        target = item["target"]["S"]
+        port = item["target"]["I"]
+
+        # Use this to give us a signal it's working
+        logger.info('{0} {1}:{2}'.format(
+            scan_id, target, port))
+
+        # TODO: Add real scan logic here to populate scan var, probably via a scan-engine
+        #  class implemented in Python instead of Ruby
```

Comment:
Yeah, Ruby => Python is a decent lift, but the value of getting it into Lambda is compelling and simplifies the engine logic when used in this context (just way less variables to handle).